src/components/TransactionHistory.js

```diff
@@ -31,34 +42,130 @@ export default function TransactionHistory() {
       );
 
       setTransactionHistory(res.data.data);
+      setFilteredTransactionHistory(res.data.data);
     };
 
     getTransactionHistory();
   }, []);
 
+  const currentHistory = filteredTransactionHistory.slice(0, transactionNumber);
+
   const handleClickShowMoreTransaction = () => {
     setTransactionNumber(transactionNumber + 5);
   };
-  const currentHistory = transactionHistory.slice(0, transactionNumber);
+
+  const handleClickSearch = () => {
+    const coinName = document.getElementById("coin-search").value;
+    const filtered = transactionHistory.filter((transaction) => {
+      if (coinName !== "") {
+        return (
+          transaction.currencyName === coinName &&
+          dates.endDate >=
+            new Date(transaction.transactionDate).setDate(
+              new Date(transaction.transactionDate).getDate() - 1
+            ) &&
+          new Date(transaction.transactionDate) >= dates.startDate
+        );
+      } else {
+        return (
+          dates.endDate >=
+            new Date(transaction.transactionDate).setDate(
+              new Date(transaction.transactionDate).getDate() - 1
+            ) && new Date(transaction.transactionDate) >= dates.startDate
+        );
+      }
+    });
+
+    setFilteredTransactionHistory(filtered);
+  };
+
+  const handleClickRefreshFilter = () => {
+    setFilteredTransactionHistory(transactionHistory);
+    setDates({ startDate: null, endDate: null });
+    document.getElementById("coin-search").value = "";
+  };
+
   return (
     <>
+      <ThemeProvider
+        theme={{
+          breakpoints: ["32em", "48em", "64em"],
+          reactDatepicker: {
+            daySize: [36, 40],
+            fontFamily: "system-ui, -apple-system",
+            colors: {
+              accessibility: "#f76f61",
+              selectedDay: "#ffcdc7",
+              selectedDayHover: "#FE8679",
+              primaryColor: "#FE8679",
+            },
+          },
+        }}
+      >
+        <DateRangeInput
+          onDatesChange={(data) => setDates(data)}
+          onFocusChange={(focusedInput) => setFocusedInput(focusedInput)}
+          startDate={dates.startDate}
+          endDate={dates.endDate}
+          focusedInput={focusedInput}
+          displayFormat="yyyy-MM-dd"
+        />
+      </ThemeProvider>
+      <input placeholder="자산구분" id="coin-search"></input>
```

Comment:
```suggestion
      <input placeholder="자산구분" id="coin-search" />
```
셀프 클로징을 할 수 있을 거 같아요 !